### DISCUSSION TITLE
_forward_unimplemented() got an unexpected keyword argument 'x' while using GNNExplainer

### DISCUSSION THREAD

**Discussion by LeavesLi1015** [pyg-team/pytorch_geometric#4576] [General]
I was planning to use GNNExplainer to explain GCN model training on MIND_small dataset.
```python
# GNNExplainer
from torch_geometric.nn import GNNExplainer
explainer = GNNExplainer(model, epochs=500, return_type='log_prob')
x, edge_index = data.x, data.edge_index
node_idx = 10
node_feat_mask, edge_mask = explainer.explain_node(node_idx, x, edge_index)
ax, G = explainer.visualize_subgraph(node_idx, edge_index, edge_mask, y=data.y)
plt.show()
```
The `model` was well trained.
But it returned `TypeError: _forward_unimplemented() got an unexpected keyword argument 'x'`.
By the way, my `data` looks like this: `Data(x=[101278, 100], edge_index=[2, 468930], weighted=[468930])`.
How do I fix this? Thx!

**Comment by rusty1s** [comment]
How is your model's `forward` function defined? It currently requires that it looks as follows:
```python
def forward(self, x, edge_index, any_other_kwargs, ...)
    pass
```

**Reply by LeavesLi1015** [reply]
I figured it out! I rewrote `forward` function and it worked! Thx a lot!
